internal/provider/common/org/edgegw.go

```diff
@@ -33,3 +40,45 @@ func (o *Org) GetEdgeGateway(egw edgegw.BaseEdgeGW) (edgegw.EdgeGateway, error)
 		NsxtEdgeGateway: govcdValues,
 	}, err
 }
+
+// GetEdgeGatewayID returns the edge gateway ID.
+func (o *Org) GetEdgeGatewayID(egw edgegw.BaseEdgeGW) (string, error) {
+	if egw.GetIDOrName() == "" {
+		return "", edgegw.ErrEdgeGatewayIDOrNameIsEmpty
+	}
+
+	if egw.GetID() != "" {
+		return egw.GetID(), nil
+	}
+
+	edgeGw, err := o.GetNsxtEdgeGatewayByName(egw.GetName())
+	if err != nil {
+		return "", err
+	}
+
+	return edgeGw.EdgeGateway.ID, nil
+}
```

Comment:
This is unnecessary. The function `GetEdgeGateway` accept `ID` or `Name` and return `edgegw.EdgeGateway`. `edgegw.EdgeGateway` has func `GetID()`